modules/oe_theme_inpage_navigation/tests/src/Kernel/Plugin/Condition/InPageNavigationStateTest.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme_inpage_navigation\Kernel\Plugin\Condition;
+
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Core\Plugin\Context\EntityContext;
+use Drupal\oe_theme_inpage_navigation\InPageNavigationHelper;
+use Drupal\Tests\node\Traits\ContentTypeCreationTrait;
+use Drupal\Tests\node\Traits\NodeCreationTrait;
+use Drupal\Tests\oe_theme\Kernel\AbstractKernelTestBase;
+
+/**
+ * Test InPageNavigationState plugin.
+ */
+class InPageNavigationStateTest extends AbstractKernelTestBase {
+
+  use ContentTypeCreationTrait;
+  use NodeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'filter',
+    'text',
+    'emr',
+    'emr_node',
+    'node',
+    'field',
+    'entity_reference_revisions',
+    'oe_theme_inpage_navigation',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('entity_meta');
+    $this->installEntitySchema('entity_meta_relation');
+    $this->installSchema('node', ['node_access']);
+
+    $this->installConfig([
+      'node',
+      'filter',
+      'emr',
+      'emr_node',
+      'oe_theme_inpage_navigation',
+    ]);
+    $this->createContentType(['type' => 'example', 'name' => 'Example']);
+    $this->container->get('emr.installer')->installEntityMetaTypeOnContentEntityType('oe_theme_inpage_navigation', 'node', ['example']);
+  }
+
+  /**
+   * Tests the inpage navigation state condition.
+   *
+   * @param bool|null $inpage_navigation_condition
+   *   Inpage navigation state condition.
+   * @param bool|null $inpage_navigation_condition_negate
+   *   Is negate.
+   * @param array|null $inpage_navigation_node_state
+   *   Inpage navigation state in node.
+   * @param array $expected
+   *   Array with expected values of 'summary' and 'result'.
+   *
+   * @dataProvider providerTestCondition
```

Comment:
This means we need to run instead of 1 kernel test, 12 kernel tests (if i counted correctly). The theme build takes 556 hours aleady to complete so we need to optimize here.

All I see the test is doing is instantiating the plugin, creating aa node and asserting the evaluation. Let's instead loop through all the cases and run them all in the same test by reinstantiating the plugin and deleting the node after the assertions.